src/Features/CSharp/Portable/Completion/CompletionProviders/CSharpDateAndTimeEmbeddedCompletionProvider.cs

```diff
@@ -7,19 +7,19 @@
 using System;
 using System.Composition;
 using Microsoft.CodeAnalysis.Completion;
-using Microsoft.CodeAnalysis.Completion.Providers;
+using Microsoft.CodeAnalysis.Features.EmbeddedLanguages.DateAndTime;
 using Microsoft.CodeAnalysis.Host.Mef;
 
 namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers
 {
-    [ExportCompletionProvider(nameof(EmbeddedLanguageCompletionProvider), LanguageNames.CSharp)]
+    [ExportCompletionProvider(nameof(CSharpDateAndTimeEmbeddedCompletionProvider), LanguageNames.CSharp)]
     [ExtensionOrder(After = nameof(ExtensionMethodImportCompletionProvider))]
     [Shared]
-    internal class EmbeddedLanguageCompletionProvider : AbstractEmbeddedLanguageCompletionProvider
+    internal class CSharpDateAndTimeEmbeddedCompletionProvider : DateAndTimeEmbeddedCompletionProvider
     {
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public EmbeddedLanguageCompletionProvider()
+        public CSharpDateAndTimeEmbeddedCompletionProvider()
```

Comment:
hrmmm.. not sure what's goign on here.  the design is to only have one compltion provide that then enumerates the embedded languages and defers to the first that match.  this way we can have embedded languages created outside of roslyn proper .